src/interfaces/phase_integrals.jl

```diff
@@ -0,0 +1,75 @@
+####################
+# The abstract phase integral interface
+#
+# In this file, the abstract interface for different copmutation methods of
+# phase integrals is defined.
+####################
+"""
+Abstract base type for defining a method for phase integral computation.
+"""
+abstract type Method end
+
+"""
+Analytic method for phase integral computation.
+
+Requires an existing implementation of analytic formulas for computing the
+entities in the phase integrals.
+"""
+struct Analytic <: Method end
+
+"""
+Fully numerical method for phase integral computation based on QuadGK.
+"""
+struct QuadGK <: Method end
+
+"""
+Struct holding setup specific information to compute phase integrals.
+
+ToDo: We mix physical and numerical aspects in this class.
+This does not seem ideal to me (Klaus).
+"""
+struct PhaseIntegral{P<:AbstractPulsedPlaneWaveField, M<:Method}
+    pulse::P
+    method::M
+end
```

Comment:
This doesn't seem like a problem to me necessarily, at least there shouldn't be any overhead since `method` is only a singleton.

This struct is intended to become the main part for user-facing functions? In this case we should think about which (if any) parts of the functionality of the  plane wave fields and the methods can be implemented as orthogonal interfaces.